src/helpers.js

```diff
@@ -4,7 +4,7 @@ export const buildNoTypenameError = node =>
   )}`
 
 export const hasField = name => set =>
```

Comment:
You are shadowing the `name` argument here so the comparison is not correct. Change the `name` argument to `fieldName` and fix the comparison.